main.cpp

```diff
@@ -41,22 +41,458 @@ If you need inspiration for what to write, take a look at previously approved st
+pastaProfit(0.0)
+{
+    cout << "construct PastaShop" << endl;
+}
+
+PastaShop::~PastaShop() 
+{
+    cout << "destruct PastaShop" << endl;
+}
+
+double PastaShop::makeCustomPasta(double newRequestedAmount, string nameOfShape  = "default", bool color1 = true, bool color2 = true, double doughWidth = 1.0)
+{
+    style.customShape(nameOfShape);
+    cout << "pasta shape requested: " << style.name << endl;
+    cout << "amount of requested pasta: " << newRequestedAmount << " lBs" << endl;
+    style.pastaColor(color1, color2);
+    style.addFlour(doughWidth);
+    cout << "you need " << style.flour << " cup of flour" << endl;
+
+    for(int i = 0; i <= newRequestedAmount; i++)
+    {
+        amountOfDoughlBS -= 1.0;
+        totalPastaWeight += 1.0;
+        cout << "total pasta sliced " << totalPastaWeight << " lBs" << endl;
+    }
+    restockDough();
+    return customRequestTotal += newRequestedAmount;
+
+}
+
+double PastaShop::restockDough()
+{
+    if (amountOfDoughlBS > 2.0)
+    {
+        cout << "you have enough dough" << endl;
+    }
+    else 
+    {
+    cout << "not enough dough, make more!!" << endl;
+
+    for(int i = 0; i < 3; i++)
+    {
+        amountOfDoughlBS += 2.0;
+        cout << "made 2 Lbs of dough" << endl;
+        cout << "you have: " << amountOfDoughlBS << " Lbs of dough total" << endl;
+    }
+    }
+    return amountOfDoughlBS;
+}
+
+void PastaShop::pastaProfitTotal()
+{
+    pastaProfit = customRequestTotal * pastaPrice;
+    cout << "profit for the hour is: $ " << pastaProfit << endl;
+}
+
 
 /*
  UDT 3:
  */
+ struct TeaParty
+ {
+    TeaParty(int cupsOfTeaAvailable_) : cupsOfTeaAvailable(cupsOfTeaAvailable_)
+    {
+        cout << "construct TeaParty" << endl;
+    }
+
+    ~TeaParty() 
+    {
+        cout << "destruct TeaParty" << endl;
+    }
+
+    int serve(int cups);    
+    bool drink(int individuals, int cups, bool food);
+    void spill();
+
+    int totalNumberOfCupsServed = 0;
+    int cupsOfTeaAvailable;
+    int participants = 3;
+    bool tooMuchTea = false;
+    bool biscuits = false;
+    bool rain = false;
+ };
+
+int TeaParty::serve(int cups)
+{
+    if(!tooMuchTea || (cupsOfTeaAvailable < 3))
+    {
+    cout << "serve " << cups << " cups of tea"<< endl;
+    totalNumberOfCupsServed = totalNumberOfCupsServed + cups;
+    cout << "total amount of tea served " << totalNumberOfCupsServed << " cups of tea"<< endl;
+    }
+    else
+        cout << "no more tea" << endl;
+    return totalNumberOfCupsServed;
+}   
+
+bool TeaParty::drink(int individuals, int cups, bool food)
+{
+    auto amountPerIndividual = cups / individuals;
+    cout << "all of the participants have had " << amountPerIndividual << " cups of tea"<< endl;
+
+    if(amountPerIndividual <= 2)
+        cout << "they can have more tea" << endl;
+    else
+    {
+        tooMuchTea = true;
+        cout << "too much tea" << endl;
+    }
+    if (food)
+        cout << "they should eat something" << endl;
+    return tooMuchTea;
```

Comment:
if you're going to use curly braces for your `if/else` blocks, use it for both parts of the `if/else` expression. 
add a blank line at the end of any `if/else` blocks so you visually see where the 'idea' ends and a new 'idea' begins (line 357)
- [ ] revise